src/compiler/transformers/es2015.ts

```diff
@@ -1571,6 +1578,78 @@ namespace ts {
                         break;
                 }
             }
+
+            /**
+             * Lazily creates the way class members will access the class prototype.
+             */
+            function getPrototypeStorageName() {
+                if (prototypeStorageName) {
+                    return prototypeStorageName;
+                }
+
+                const originalPrototypeAccess = createPropertyAccess(getInternalName(node), "prototype");

```

Comment:
This needs to be updated to use the new `NodeFactory` API introduced in 4.0.:

```suggestion
                const originalPrototypeAccess = factory.createPropertyAccess(factory.getInternalName(node), "prototype");
```

Both here and in other places.